### ISSUE TITLE
Loop error causes that no row is rendered

### ISSUE THREAD

**Issue by ScripterYoda** [olifolkerd/tabulator#3070]
**Describe the bug**
No rows or error message error is rendered, just headers and footer. In web browser console, there is an error that a call stack size has been excedeed and that functions adjustTableSize and redraw where calling each other in a loop

**Tabulator Info**
4.8.2

**Working Example**
I am unable to create an example as the issue appears randomly and disappears after restarting computer. Therefore, I have not been able to reproduce it, but clients report it from time to time and it is breaking.

**To Reproduce**
Client just opens a page in which javascript generates a table.

**Expected behavior**
To rows in table to be generated.

**Screenshots**
Error stack:
![image](https://user-images.githubusercontent.com/8488112/98609509-b12c5c00-22ed-11eb-814e-13f1ec2007d2.png)
Rendered table:
![image](https://user-images.githubusercontent.com/8488112/98610090-20568000-22ef-11eb-9eee-3a6cde611cd8.png)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome
 - Version: 86

**Smartphone (please complete the following information):**
N/A

**Additional context**
Code used to create a table:

```
table = new Tabulator(tid, {
        locale:true,
      	history:true,
      	placeholder:"Brak danych",
        layout:"fitDataFill",
        responsiveLayout:"collapse",
        responsiveLayoutCollapseStartOpen:false,
        movableColumns: true,
        pagination:"remote",
        footerElement:"...",
        ajaxSorting:true,
        ajaxParams:settings.ajaxParams,
        paginationSize:settings.paginationSize,
        paginationSizeSelector:[10,50,100,250],
        langs: "...",
        ajaxURL:settings.urlPrefix + "/data.json",
        columns:columns,
        initialSort:settings.initialSort,
        cellEdited:function(cell){
        	if (settings.ajaxUpdateDataFunction != null) {
        	$.ajax({
       		  type: "POST",
       		  dataType: "text",
       		  url: settings.urlPrefix + "/dataUpdate.json",
       		  data: settings.ajaxUpdateDataFunction(cell),
       		  error: function (request, status, error) {
     	   		  table.undo();
     	   		  alert("Edycja pola nie powiodĹa siÄ.")
       		  }
       		});
        	}
        },
        tableBuilt:function(){
            $(tid).addClass("thead-dark table-bordered");
        }
        
    });
```

**Comment by Eixix** [comment]
Having this issue as well, it seems that this is tied to specific zoom options and resolutions in the browser (also altered with the console open). Changing the resolution and reloading the page works for me.

**Comment by ScripterYoda** [comment]
Ha! You're right. I am finally able to reproduce the error. It appears when I have the size of page set to 110%.

**Comment by olifolkerd** [comment]
If you have a way to reproduce the error please post a link to a demo, I have been looking to fix it for weeks but no one seems to be able to produce a demo that reliably fails so I have no was to debug it.

Cheers

Oli

**Comment by olifolkerd** [comment]
Hey @ScripterYoda @Eixix 

I there any chance that you could post a link to a JS Fiddle, Code Pen or any other example that demonstrates the issue. i desperately want to get to the bottom of this issue, but am completely unable to replicate the issue in any way on my test setup. numerous people have raised the issue but know one can show me how to reproduce it in a simple way.

The code you posted above sadly doesn't help as it contains too many references to data outside the table so there is no way for me to replicate it.

Please if you have 10 mins to spare in the next couple of days could you put an example together that i could test against. i am planning on doing the 4.9 version release next weekend and it would be amazing if i could include the fix in that.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
@ScripterYoda and i think i now understand the issue, it is related to the table being in a fixed height container when pagination is enabled and with a number of rows set that exceeds the height of the table with a zoom of not 100%, and only on certain Tabulator themes.

A bit neiche but now i have a repeatable example i can fix it. It was the hint to applying the bootstrap themes in your table built function that did it. By the way it would be better to apply them in the ***tableBuilding*** callback that way when the table is laying out rows it can correctly calculate the row height of that theme.

Cheers

Oli :)

**Comment by olifolkerd** [comment]
Hey @ScripterYoda 

I have just pushed a fix for this to the 4.9 branch that will be release later this month. Thanks for the hint towards the themes, that was the last bit of information i needed to get to the bottom of things *(i hope)*

Cheers

Oli :)
